test/csuite/wt9199_checkpoint_txn_commit_race/main.c

```diff
@@ -0,0 +1,274 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * This test is to create a window where a checkpoint can fail to include a transaction's updates
+ * with commit times before stable. Added a WT_TIMING_STRESS_COMMIT_TRANSACTION_SLOW timing stress
+ * to add a 10 second delay while committing a transaction.
+ */
+
+#define NUM_RECORDS WT_THOUSAND
+
+/* Constants and variables declaration. */
+
+static const char conn_config[] =
+  "create,cache_size=2GB,statistics=(all),statistics_log=(json,on_close,wait=1),,timing_stress_for_"
+  "test=[commit_transaction_slow]";
+static const char table_config_row[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=Q,value_format=Q";
+static const char *const uri = "table:wt9199-checkpoint-txn-commit-race";
+static uint64_t global_stable_ts;
+
+static bool inserted;
```

Comment:
```suggestion
static volatile bool inserted;
```

Reply:
Fixed in 746beb4a74be524a787cd7d49ab66860f4b9617c